tools/m2kcli/commands/communication/uart.cpp

```diff
@@ -0,0 +1,193 @@
+			case 'r':
+				handleRead();
+				break;
+			default:
+				break;
+		}
+	}
+	return false;
+}
+
+void Uart::handleInit()
+{
+	std::map<std::string, std::string> arguments = Validator::validate(getArguments());
+	if (!(arguments.count("device") && arguments.count("baud_rate") && arguments.count("parity") &&
+	      arguments.count("bits_number") && arguments.count("stop_bits"))) {
+		throw std::runtime_error(
+			"Expecting: device=<index> baud_rate=<value> parity=<value> bits_number=<value> stop_bits=<value>\n");
+	}
+
+	int device;
+	Validator::validate(arguments["device"], "device", device);
+
+	int baud_rate;
+	Validator::validate(arguments["baud_rate"], "baud_rate", baud_rate);
+
+	int bits_number;
+	Validator::validate(arguments["bits_number"], "bits_number", bits_number);
+
+	std::string stop_bits;
+	Validator::validate(arguments["stop_bits"], "stop_bits", stop_bits);
+	int enumIndexStop = getIndexOfElement(stop_bits, this->stop_bits);
+	if (enumIndexStop == -1) {
+		throw std::runtime_error("Invalid number of stop bits: '" + std::string(stop_bits) + "'.\n");
+	}
+
+	std::string parity;
+	Validator::validate(arguments["parity"], "parity", parity);
+	int enumIndexParity = getIndexOfElement(parity, this->parity);
+	if (enumIndexParity == -1) {
+		throw std::runtime_error("Invalid parity: '" + std::string(parity) + "'.\n");
+	}
+	m2k_uart_init m2KUartInit;
+	m2KUartInit.bits_number = bits_number;
+	m2KUartInit.parity = static_cast<uart_parity>(enumIndexParity);
+	m2KUartInit.stop_bits = static_cast<uart_stop_bits>(enumIndexStop + 2);
+	m2KUartInit.context = context;
+
+	uart_init_param uartInitParam;
+	uartInitParam.device_id = 0;
+	uartInitParam.baud_rate = 9600;
+	uartInitParam.extra = (void *) &m2KUartInit;
+
+	if (uart_init(&uartDesc, &uartInitParam) == -1) {
+		throw std::runtime_error("Could not initiate UART\n");
+	}
+}
+
+void Uart::handleWrite()
+{
+	std::map<std::string, std::string> arguments = Validator::validate(getArguments());
+	if (!(arguments.count("data"))) {
+		throw std::runtime_error("Expecting: data=<value>,...\n");
+	}
+
+	std::vector<uint8_t> data;
+	Validator::validate(arguments["data"], "data", data);
+
+	uart_write(uartDesc, data.data(), data.size());
+}
+
+void Uart::handleRead()
+{
+	std::map<std::string, std::string> arguments = Validator::validate(getArguments());
+	if (!(arguments.count("bytes_number") && arguments.count("device"))) {
+		throw std::runtime_error("Expecting: device=<index> bytes_number=<value> format=<value>\n");
+	}
+
+	int device;
+	Validator::validate(arguments["device"], "device", device);
+
+	int bytesNumber;
+	Validator::validate(arguments["bytes_number"], "bytes_number", bytesNumber);
+
+	std::string format;
+	if (arguments.count("format")) {
+		Validator::validate(arguments["format"], "format", format);
+	}
+
+	uartDesc->device_id = device;
+	std::vector<uint8_t> data(bytesNumber);
+	uart_read(uartDesc, data.data(), bytesNumber);
+
+	if (format.empty() || format == "text") {
```

Comment:
Is it possible to create an option/format that gets data from stdin ?

Reply:
If the data content is empty (e.g. "--write date=") the data will be read from stdin.